inst/htmljs/animint.js

```diff
@@ -1626,11 +1630,51 @@ var animint = function (to_select, json_file) {
 	  return d["clickSelects.variable"] + " " + d["clickSelects.value"];
 	};
       }
-      // if elements have an existing title, remove it.
-      elements.selectAll("title").remove();
-      elements.append("svg:title")
-        .text(get_fun(text_fun))
-      ;
+      var tooltip = d3.select("#plot").select(".animint-tooltip").node() 
+    ? d3.select(".animint-tooltip")
+    : d3.select("#plot").append("div")
+        .attr("class", "animint-tooltip")
+        .style("opacity", 0);
+      // Add tooltip handlers
+      elements
+        .on("mouseover.tooltip", function(d) {
+          if (!d || typeof text_fun !== 'function') return;
+          var content;
+          try {
+            content = text_fun(d);
+          } catch(e) {
+            console.error("Error generating tooltip content:", e);
+            return;
+          }
+          var mouseX = 0, mouseY = 0;
+          try {
+            mouseX = (d3.event && d3.event.pageX) || 0;
+            mouseY = (d3.event && d3.event.pageY) || 0;
+          } catch(e) {
+            console.error("Error getting mouse position:", e);
+          }
+          tooltip
+            .html(content)
+            .style("left", (mouseX + TOOLTIP_HORIZONTAL_OFFSET) + "px")
+            .style("top", (mouseY - TOOLTIP_VERTICAL_OFFSET) + "px")
+            .style("opacity", 1);
```

Comment:
this block is almost repeated below.
can you please make a function that is used twice, in order to emphasize the similarity?